Python/Source/Pick.cpp

```diff
@@ -0,0 +1,32 @@
+#include <pybind11/functional.h>
+#include "Autobox.hpp"
+#include "Pick.hpp"
+
+using namespace pybind11;
+using namespace Rover;
+
+namespace Rover {
+  void export_pick(module& module) {
+    class_<PythonPick>(module, "Pick")
+      .def(init<object, args>())
+      .def("generate", &PythonPick::generate);
+    implicitly_convertible<PythonPick, Box<object>>();
+  }
+
+  PythonPick::PythonPick(object choice, args generators)
+    : m_choice(python_autobox<std::size_t, object>(std::move(choice))),
+      m_generators([&] {
+        auto result = std::vector<Box<object>>();
+        for(auto i = std::size_t(0); i < generators.size(); ++i) {
+          result.push_back(python_autobox<object, object>(std::move(
+            generators[i])));
+        }
+        return result;
+      }()) {}
+
+  PythonPick::Type PythonPick::generate(Evaluator& evaluator) {
+    auto index = evaluator.evaluate(m_choice);
```

Comment:
Actually, `python_autobox<std::size_t, ...>` is designed to do exactly that: evaluate the generator and then cast it to `std::size_t`. So this logic is already encapsulated inside `autobox`. I could use `Box<object>` and then explicit cast too, but it does the same thing.

Reply:
Okay that's fine then. Prefer ```int``` instead of ```size_t``` for this branch and in general. We almost never use ```size_t``` unless we're dealing with very low level stuff.